src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1123,7 +1123,9 @@ private async Task ReceiveLoop(ConnectionState connectionState)
             var uploadStreamSource = new CancellationTokenSource();
             connectionState.UploadStreamToken = uploadStreamSource.Token;
             var invocationMessageChannel = Channel.CreateUnbounded<InvocationMessage>(_receiveLoopOptions);
-            var invocationMessageReceiveTask = StartProcessingInvocationMessages(invocationMessageChannel.Reader);
+
+            // We can't safely wait for this task when closing without introducing deadlock potential when calling StopAsync in a .On method
+            _ = StartProcessingInvocationMessages(invocationMessageChannel.Reader);
```

Comment:
Can we store it as a field on connectionState instead? That way we'll be able to look at it in dumps.

Reply:
Sounds reasonable